src/nvim/api/buffer.c

```diff
@@ -636,6 +634,297 @@ ArrayOf(Integer, 2) buffer_get_mark(Buffer buffer, String name, Error *err)
   return rv;
 }
 
+/// Returns mark info at the given position or mark index
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with namespace_create
+/// @param id [row, col] or mark_id of mark
+/// @param[out] err Details of an error that may have occurred
+/// @return [mark_id, row, col]
+ArrayOf(Object) buffer_mark_index(Buffer buffer, Integer namespace, Object id, Error *err)
```

Comment:
They will be confused in either case. It's confusing to have inconsistent convention.  

`buf_get_mark_index` is at least a bit more "discoverable" since it will appear immediately next to `buf_get_mark` in documentation, code-completion, API function list, etc.